File: 2024/Days/1-5/1/Code/Php/part1Solution.php
<?php

$input1 = file_get_contents("input1.txt");

$lines = explode("\n", $input1);
while($lines[sizeof($lines) - 1] == ""){
    //Remove all empty strings found starting from the end of the array.
    array_pop($lines);
}

$column1 = [];
$column2 = [];
for($i = 0; $i < sizeof($lines); $i++){
    //Separate each line by space.
    $fields = explode(' ', $lines[$i]);

    //Since we do not know how many spaces between each value, we take the first and last element.
    $value1 = $fields[0];
    $value2 = $fields[sizeof($fields) - 1];
    
    array_push($column1, intval($value1));
    array_push($column2, intval($value2));
}

sort($column1);
sort($column2);

$sum = 0;
for($i = 0; $i < sizeof($column1); $i++){
    $sum += abs($column1[$i] - $column2[$i]);
}

echo $sum . "\n";

?>

File: 2024/Days/1-5/1/Code/Python/part1Solution.py
import os, re


def main():
    try:
        current_dir = os.getcwd()
        with open(f"{current_dir}/input1.txt", 'r') as input1_file:
            input1 = input1_file.read()
            lines = input1.split('\n')
            lines.remove('')

            def to_tuple(line) -> tuple[int, int]:
                splitted_line = re.sub(' +', ' ', line).split(' ')
                return int(splitted_line[0]), int(splitted_line[1])

            table = [to_tuple(line) for line in lines]

            column1 = [column1 for column1, _ in table]
            column1.sort()

            column2 = [column2 for _, column2 in table]
            column2.sort()

            table = list(zip(column1, column2))

            total_sum = sum([abs(column1 - column2) for column1, column2 in table])
            print(total_sum)
        
    except Exception as error:
        print(error)

if __name__ == "__main__":
    main()
